src/expr/src/vector_op/timestamptz.rs

```diff
@@ -62,6 +65,39 @@ pub fn timestamp_at_time_zone(input: NaiveDateTimeWrapper, time_zone: &str) -> R
     Ok(usec)
 }
 
+pub fn timestamptz_to_string(elem: i64, time_zone: &str, writer: &mut dyn Write) -> Result<()> {
+    let time_zone = lookup_time_zone(time_zone)?;
+    let secs = elem.div_euclid(1_000_000);
+    let nsecs = elem.rem_euclid(1_000_000) * 1000;
+    let instant_utc = Utc.timestamp_opt(secs, nsecs as u32).unwrap();
+    let instant_local = instant_utc.with_timezone(&time_zone);
+    // TODO: err if not divisible by 3600 seconds?
+    let hours_offset = instant_local.offset().fix().local_minus_utc() / 3600;
+    if hours_offset >= 0 {
+        write!(
+            writer,
+            "{}+{:0>2}:00",
+            instant_local.naive_local(),
+            hours_offset
+        )
+        .unwrap();
+    } else {
+        write!(
+            writer,
+            "{}-{:0>2}:00",
+            instant_local.naive_local(),
+            -hours_offset
+        )
+        .unwrap();
+    }
+    Ok(())
+}
+
+pub fn str_to_timestamptz(elem: &str, time_zone: &str) -> Result<i64> {
+    fallible_str_to_timestamptz(elem)
```

Comment:
Why call `fallible_str_to_timestamptz(elem)` first?

Reply:
`fallible_str_to_timestamptz` handles `2022-01-01 12:00:00-08:00`, where timezone is provided by value itself rather than session. If that fails, this tries to interpret the value (without timezone) in the session timezone.

Reply:
Got it. Perhaps need a comment.

Reply:
Yes, the naming should also be better. `str_with_timezone_to_timestamptz(elem)`